File: Projet_SHAITA_DRIOUCHE_INFO0503/serveur_JAVA/src/package_java/Commande.java
package package_java;

import java.io.Serializable;
import org.json.JSONObject;

import java.text.DateFormat;
import java.text.SimpleDateFormat;
import java.util.Date;

/**
 * classe permettant de reconstruire la commande du client
 * 
 * @author Sami DRIOUCHE & Fayssal SHAITA
 */
public class Commande implements Serializable {

    // Attributs
    private Date date;
    private String typeEnergie;
    private double quantiteDesire;
    private String modeExtraction;
    private String origineGeographique;
    private double prixMaxUnite; // prix maximum par l'unité de l'energie
    private int idClient;// l'utilisateur
    private String etat;// l'etat de la commande
    private CodeSuivi code; // code de suivi de la commande

    // constructeur
    public Commande(String typeEnergie, double quantiteDesire, String modeExtraction, String origineGeographique,
            double prixMaxUnite, int idClient, String etat, Date date) {
        this.typeEnergie = typeEnergie;
        this.quantiteDesire = quantiteDesire;
        this.modeExtraction = modeExtraction;
        this.origineGeographique = origineGeographique;
        this.prixMaxUnite = prixMaxUnite;
        this.idClient = idClient;
        this.etat = etat;
        this.date = date;
        this.code = new CodeSuivi(typeEnergie, prixMaxUnite, quantiteDesire, modeExtraction, origineGeographique,
                idClient, etat);
    }

    // getter
    public Date getDate() {
        return date;
    }

    public String getTypeEnergie() {
        return typeEnergie;
    }

    public double getQuantiteDesire() {
        return quantiteDesire;
    }

    public String getModeExtraction() {
        return modeExtraction;
    }

    public String getOrigineGeographique() {
        return origineGeographique;
    }

    public double getPrixMaxUnite() {
        return prixMaxUnite;
    }

    public int getIdClient() {
        return idClient;
    }

    public String getEtat() {
        return etat;
    }

    public CodeSuivi getCode() {
        return code;
    }

    // setter
    public void setDate(Date date) {
        this.date = date;
    }

    public void setTypeEnergie(String typeEnergie) {
        this.typeEnergie = typeEnergie;
    }

    public void setQuantiteDesire(double quantiteDesire) {
        this.quantiteDesire = quantiteDesire;
    }

    public void setModeExtraction(String modeExtraction) {
        this.modeExtraction = modeExtraction;
    }

    public void setOrigineGeographique(String origineGeographique) {
        this.origineGeographique = origineGeographique;
    }

    public void setPrixMaxUnite(double prixMaxUnite) {
        this.prixMaxUnite = prixMaxUnite;
    }

    public void setIdClient(int idClient) {
        this.idClient = idClient;
    }

    public void setEtat(String etat) {
        this.etat = etat;
    }

    public void setCode(CodeSuivi code) {
        this.code = code;
    }

    // toString
    @Override
    public String toString() {
        return "\ntype d'energie :" + typeEnergie + " \nquantitee desire :" + quantiteDesire + " \nmode d'extraction : "
                + modeExtraction
                + "\norigine geographique : " + origineGeographique + "\nprix max par unite : " + prixMaxUnite
                + "\nid client :" + idClient
                + " \netat : " + etat;
    }

    // tojson
    public JSONObject toJSON() {
        JSONObject json = new JSONObject();
        DateFormat format = new SimpleDateFormat("dd-MM-yyyy");
        json.put("date", format.format(date));
        json.put("typeEnergie", typeEnergie);
        json.put("quantiteDesire", quantiteDesire);
        json.put("modeExtraction", modeExtraction);
        json.put("origineGeographique", origineGeographique);
        json.put("prixMaxUnite", prixMaxUnite);
        json.put("idClient", idClient);
        json.put("etat", etat);
        return json;
    }

    // fromjson
    public static Commande fromJSON(String json) {
        JSONObject Objet = new JSONObject(json);
        DateFormat format = new SimpleDateFormat("dd-MM-yyyy");
        Date date = null;
        try {
            date = format.parse(Objet.getString("date"));
        } catch (Exception e) {
            e.printStackTrace();
        }

        Commande commande = new Commande(Objet.getString("typeEnergie"), Objet.getDouble("quantiteDesire"),
                Objet.getString("modeExtraction"), Objet.getString("origineGeographique"),
                Objet.getDouble("prixMaxUnite"),
                Objet.getInt("idClient"), Objet.getString("etat"), date);
        return commande;
    }
}

File: Projet_SHAITA_DRIOUCHE_INFO0503/serveur_PHP/package_php/commande.php
<?php

class Commande implements JsonSerializable{

    // Attributs
    private $date;   
    private string $typeEnergie; //petrole hydrogene electricite
    private float $quantiteDesire;
    private string $modeExtraction;
    private string $origineGeographique;
    private float $prixMaxUnite; // prix maximum par l'unité de l'energie
    private int $idClient ;//l'utilisateur
    private string $etat ;//l'etat de la commande

    // Constructeur
    public function __construct( string $typeEnergie,float $quantiteDesire ,string $modeExtraction  ,string $origineGeographique  ,float $prixMaxUnite, int $idClient, string $etat ){
        $this->date = date("d-m-Y");      
        $this->typeEnergie = $typeEnergie;
        $this->quantiteDesire = $quantiteDesire;
        $this->modeExtraction = $modeExtraction;
        $this->origineGeographique = $origineGeographique;
        $this->prixMaxUnite = $prixMaxUnite;
        $this->idClient = $idClient;
        $this->etat = $etat;
        $this->verifTypeEnergie();
        $this->verifOrigineGeographique();
        $this->verifModeExtraction();
        
    }

    // Getters
    public function getDate() {
        return $this->date;
    }

    public function getTypeEnergie() {
        return $this->typeEnergie;
    }

    public function getQuantiteDesire() {
        return $this->quantiteDesire;
    }

    public function getModeExtraction() {
        return $this->modeExtraction;
    }

    public function getOrigineGeographique() {
        return $this->origineGeographique;
    }

    public function getPrixMaxUnite() {
        return $this->prixMaxUnite;
    }

    public function getIdClient() {
        return $this->idClient;
    }

    public function getEtat() {
        return $this->etat;
    }

    // Setters
    public function setDate($date) {
        $this->date = $date;
    }

    public function setTypeEnergie($typeEnergie) {
        $this->typeEnergie = $typeEnergie;
    }

    public function setQuantiteDesire($quantiteDesire) {
        $this->quantiteDesire = $quantiteDesire;
    }

    public function setModeExtraction($modeExtraction) {
        $this->modeExtraction = $modeExtraction;
    }

    public function setOrigineGeographique($origineGeographique) {
        $this->origineGeographique = $origineGeographique;
    }

    public function setPrixMaxUnite($prixMaxUnite) {
        $this->prixMaxUnite = $prixMaxUnite;
    }

    public function setIdClient($idClient) {
        $this->idClient = $idClient;
    }

    public function setEtat($etat) {
        $this->etat = $etat;
    }

    

    public function jsonSerialize() : array {
        $json = array();
        $json['date'] = $this->date;
        $json['typeEnergie'] = $this->typeEnergie;
        $json['quantiteDesire'] = $this->quantiteDesire;
        $json['modeExtraction'] = $this->modeExtraction;
        $json['origineGeographique'] = $this->origineGeographique;
        $json['prixMaxUnite'] = $this->prixMaxUnite;
        $json['idClient'] = $this->idClient;
        $json['etat'] = $this->etat;
        return $json;
    }

    public static function fromJson(string $json) : Commande {     
        $data = json_decode($json, true);
        $commande = new Commande($data['typeEnergie'], $data['quantiteDesire'], $data['modeExtraction'], $data['origineGeographique'], $data['prixMaxUnite'], $data['idClient'],$data['etat']);
        return $commande;
    }


    public function verifTypeEnergie(){
       switch($this->typeEnergie){
           case "PETROLE":
            $this->typeEnergie = "Petrole";
            break;
            case "HYDROGENE":
            $this->typeEnergie = "Hydrogene";
            break;
            case "ELECTRICITE":
            $this->typeEnergie = "Electricite";
            break;
            case "BIOMETHANE":  
            $this->typeEnergie = "Biomethane";
            break;
            case "GAZNATUREL":
            $this->typeEnergie = "Gaz naturel";
            break;
       }
    }

    public function verifModeExtraction(){
        switch($this->modeExtraction){
            case "EOLIEN":
             $this->modeExtraction = "Eolien";
             break;
             case "NUCLEAIRE":
             $this->modeExtraction = "Nucleaire";
             break;
             case "FORAGEVERTICAL":
             $this->modeExtraction = "Forage Vertical";
             break;
             case "FORAGEHORIZONTAL":  
             $this->modeExtraction = "Forage Horizontal";
                break;
             case "SOLAIRE":
             $this->modeExtraction = "Solaire";
                break;
            
        }
    }

    public function verifOrigineGeographique(){
        switch($this->origineGeographique){
            case "FRANCE":
                $this->origineGeographique = "France";
                break;
            case "ESPAGNE":
                $this->origineGeographique = "Espagne";
                break;
            case "ALLEMAGNE":
                $this->origineGeographique = "Allemagne";
                break;
            case "ITALIE":  
                $this->origineGeographique = "Italie";
                break;
            case "ROYAUMEUNI":
                $this->origineGeographique = "Royaume-Uni";
                break;
            case "CHINE": 
                $this->origineGeographique = "Chine";
                break;
            case "PAYSBAS": 
                $this->origineGeographique = "Pays-Bas";
                break;
            case "PORTUGAL": 
                $this->origineGeographique = "Portugal";
                break;
            case "SUEDE": 
                $this->origineGeographique = "Suede";
                break;
            case "FINLANDE": 
                $this->origineGeographique = "Finlande";
                break;
            case "ALGERIE": 
                $this->origineGeographique = "Algerie";
                break;
            case "MAROC":
                $this->origineGeographique = "Maroc";
                break;
            case "TUNISIE":
                $this->origineGeographique = "Tunisie";
                break;
            case "TURQUIE":
                $this->origineGeographique = "Turquie";
                break;

            
        }
    }
}

?>